app/services/apiQueues.js

```diff
@@ -12,12 +12,30 @@ function queues(queueFactory) {
 }
 
 function exec(message) {
-  request(message.options, function(error, response, body){
-    if (error){
-      console.log(error);
-      message._reject(error);
+  const {
+    options: {
+      method,
+      url
+    }
+  } = message;
+
+  fetch(url, message.options).then(function(response) {
```

Comment:
Because it's a browser API, it's likely implemented in the code of the javascript engine. because this mostly makes http requests and gives you flexibility into how the responses are parsed, I'd guess this is approximately as efficient as the previous implementation. In practice, I've found that analyzing a large folder and uploading many things is a pretty fast operation.